src/main/java/picard/illumina/CheckIlluminaDirectory.java

```diff
@@ -161,42 +165,44 @@ protected int doWork() {
 
                 //check s.locs
                 final File locsFile = new File(BASECALLS_DIR.getParentFile(), AbstractIlluminaPositionFileReader.S_LOCS_FILE);
-                final LocsFileReader locsFileReader = new LocsFileReader(locsFile);
-                final List<AbstractIlluminaPositionFileReader.PositionInfo> locs = new ArrayList<>();
-                while (locsFileReader.hasNext()) {
-                    locs.add(locsFileReader.next());
+                final List<AbstractIlluminaPositionFileReader.PositionInfo> locs;
+                final Map<Integer, File> filterFileMap;
+                try (LocsFileReader locsFileReader = new LocsFileReader(locsFile)) {
+                    locs = new ArrayList<>();
+                    while (locsFileReader.hasNext()) {
+                        locs.add(locsFileReader.next());
+                    }
                 }
 
-                final Map<Integer, File> filterFileMap = new HashMap<>();
+                filterFileMap = new HashMap<>();
```

Comment:
Same question, why not assign at the point where you declare it?

Reply:
I'm not sure why these got changed. I didn't change them... I least I don't remember changing them.